test/units/module_utils/basic/test_run_command.py

```diff
@@ -41,38 +39,112 @@ def mock_os_abspath(path):
             return os.getcwd.return_value + '/' + path
 
     os = mocker.patch('ansible.module_utils.basic.os')
-    os._cmd_out = {
-        # os.read() is returning 'bytes', not strings
-        mocker.sentinel.stdout: BytesIO(),
-        mocker.sentinel.stderr: BytesIO(),
-    }
 
     os.path.expandvars.side_effect = lambda x: x
     os.path.expanduser.side_effect = lambda x: x
     os.environ = {'PATH': '/bin'}
     os.getcwd.return_value = '/home/foo'
     os.path.isdir.return_value = True
     os.chdir.side_effect = mock_os_chdir
-    os.read.side_effect = mock_os_read
     os.path.abspath.side_effect = mock_os_abspath
 
     yield os
 
 
+class DummyFileObj():
+    def __init__(self, fileobj):
+        self.fileobj = fileobj
+
+
+class MockSelector(selectors.BaseSelector):
+    def __init__(self):
+        super(MockSelector, self).__init__()
+        self._fds = []
+
+    def register(self, fileobj, events, data=None):
+        self._fds.append(fileobj)
+        return 1
+
+    def unregister(self, fileobj):
+        self._fds.remove(fileobj)
+
+    def select(self, timeout=None):
+        ready = []
+        for fd in self._fds:
+            ready.append((DummyFileObj(fd), selectors.EVENT_READ))
+        return ready
+
+    def get_map(self):
+        return self._fds
+
+    def close(self):
+        super(MockSelector, self).close()
+        self._fds = []
+
+
+class SpecialBytesIO(BytesIO):
+    def __init__(self, *args, **kwargs):
+        fh = kwargs.pop('fh', None)
+        super(SpecialBytesIO, self).__init__(*args, **kwargs)
+        self.fh = fh
+
+    def fileno(self):
+        return self.fh
+
+    # We need to do this because some of our tests create a new value for stdout and stderr
+    # The new value is able to affect the string that is returned by the subprocess stdout and
+    # stderr but by the time the test gets it, it is too late to change the SpecialBytesIO that
+    # subprocess.Popen returns for stdout and stderr.  If we could figure out how to change those as
+    # well, then we wouldn't need this.
+    def __eq__(self, other):
+        if id(self) == id(other) or self.fh == other.fileno():
+            return True
+        return False
+
+
+class DummyKey:
+    def __init__(self, fileobj):
+        self.fileobj = fileobj
+
+
 @pytest.fixture
 def mock_subprocess(mocker):
-    def mock_select(rlist, wlist, xlist, timeout=1):
-        return (rlist, [], [])
 
-    fake_select = mocker.patch('ansible.module_utils.basic.select')
-    fake_select.select.side_effect = mock_select
+    class MockSelector(selectors.BaseSelector):
```

Comment:
The pylint error is occurring because this `MockSelector` class shadows the one on line 59.